src/atomate2/vasp/jobs/mp.py

```diff
@@ -0,0 +1,158 @@
+"""
+Module defining jobs for Materials Project r2SCAN workflows.
+
+Reference: https://doi.org/10.1103/PhysRevMaterials.6.013801
+"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING
+
+import numpy as np
+from monty.serialization import loadfn
+from pkg_resources import resource_filename
+
+from atomate2.vasp.jobs.base import BaseVaspMaker
+from atomate2.vasp.jobs.core import StaticSetGenerator
+from atomate2.vasp.sets.base import VaspInputGenerator
+
+if TYPE_CHECKING:
+    from pymatgen.core import Structure
+
+__all__ = ["MPPreRelaxMaker", "MPRelaxMaker", "MPStaticMaker"]
+
+_BASE_MP_R2SCAN_RELAX_SET = loadfn(
+    resource_filename("atomate2.vasp.sets", "BaseMPr2SCANRelaxSet.yaml")
+)
+
+
+class MPRelaxR2SCANGenerator(VaspInputGenerator):
+    config_dict: dict = _BASE_MP_R2SCAN_RELAX_SET
+
+
+@dataclass
+class MPPreRelaxMaker(BaseVaspMaker):
+    """
+    Maker to create VASP pre-relaxation job using PBEsol by default.
+
+    Parameters
+    ----------
+    name : str
+        The job name.
+    input_set_generator : .VaspInputGenerator
+        A generator used to make the input set.
+    write_input_set_kwargs : dict
+        Keyword arguments that will get passed to :obj:`.write_vasp_input_set`.
+    copy_vasp_kwargs : dict
+        Keyword arguments that will get passed to :obj:`.copy_vasp_outputs`.
+    run_vasp_kwargs : dict
+        Keyword arguments that will get passed to :obj:`.run_vasp`.
+    task_document_kwargs : dict
+        Keyword arguments that will get passed to :obj:`.TaskDoc.from_directory`.
+    stop_children_kwargs : dict
+        Keyword arguments that will get passed to :obj:`.should_stop_children`.
+    write_additional_data : dict
+        Additional data to write to the current directory. Given as a dict of
+        {filename: data}. Note that if using FireWorks, dictionary keys cannot contain
+        the "." character which is typically used to denote file extensions. To avoid
+        this, use the ":" character, which will automatically be converted to ".". E.g.
+        ``{"my_file:txt": "contents of the file"}``.
+    """
+
+    name: str = "MP PreRelax"
+    input_set_generator: VaspInputGenerator = field(
```

Comment:
Should the name be more descriptive, e.g. "MP PBESol PreRelax"?

Reply:
I'm on the fence about this. People might forget to customize the name if they change to, say, ML pre-relaxation in which case "MP PBESol PreRelax" would be misleading. I guess the job can try to be smart about it and set the name based on the `InputGenerator`. @utf What's `atomate2`'s general policy about this?

Speaking of which, since `atomate2` has 1st class ML pre-relax support now from @matthewkuner's excellent #322, is the type hint `VaspInputGenerator` on the pre-relax job sufficiently generic? Doesn't look like `class CHGNetRelaxMaker(Maker)` is compatible with `VaspInputGenerator` .

Reply:
I agree with @janosh on this. We don't want to tie the name to a given setting to allow for users to override things if they want to inherit the MP workflow but not use it as-is.

Reply:
@janosh do you think we should try to make CHGNetRelaxMaker compatible with the VaspInputGenerator? I wasn’t sure how to go about that when I was writing the code initially

Reply:
Regarding @janosh's point, I think the type hint itself is okay. After all, the user needs to use a pre-relax step that outputs a bandgap, and that's going to be impossible to get from `CHGNetRelaxMaker`.

Regarding your question about if it should be made compatible with `VaspInputGenerator`, I would lean towards no. It's probably best to keep it distinct since not all the fields in the schema can be populated from CHGNet.

Reply:
I think `CHGNetRelaxMaker` is good the way it is and inherently different from `VaspInputGenerator`. My comment this morning was confused/not fully awake yet. 😄  `MPPreRelaxMaker` can stay the way it is. You would simply swap it out entirely in `MP2023RelaxMaker` if you want to use `CHGNetRelaxMaker`.